### DISCUSSION TITLE
How to get system accent color

### DISCUSSION THREAD

**Discussion by koohz** [tauri-apps/tauri#5305] [Q&A]
How i can get the system accent color in Windows?
In the `windows` package i've found `windows::Win32::Graphics::Dwm::DwmGetColorizationColor` but i'm not fluent with rust so basically i don't know how i can call it

Btw, I don't know if it could be a feature request or something that exists but it's not documented

**Comment by FabianLars** [comment]
Tauri does not have an API for that yet. It's being worked on for macOS at least: https://github.com/tauri-apps/tao/pull/589

A rust based solution would look something like this (using `windows = "0.39"` and for simplicity `hex = "0.4"`
```rs
    let mut colorization: u32 = 0;
    let mut opaqueblend = windows::Win32::Foundation::BOOL(0);
    unsafe {
        windows::Win32::Graphics::Dwm::DwmGetColorizationColor(&mut colorization, &mut opaqueblend)
    }; // this can return an error

    let argb = hex::decode(format!("{:X}", colorization)).unwrap(); // this too, so don't use unwrap if you don't wanna see your app crashing.
```

**Comment by amrbashir** [comment]
This is another solution that uses `windows-rs` crate and gets the correct accent color even when using "Automatic Accent Color" option in the Windows settings.
```rs
pub fn system_accent_color() -> Option<String> {
    let settings = UISettings::new().ok()?;
    // You can change `UIColorType` variant to get different shading
    let color = settings.GetColorValue(UIColorType::Accent).ok()?;
    let color_rgb = format!("rgba({},{},{},{})", color.R, color.G, color.B, color.A);
    Some(color_rgb)
}
```
